nemo/core/classes/modelPT.py

```diff
@@ -325,6 +346,169 @@ def test_dataloader(self):
+            This default can be changed by passing the special key `test_loss_idx: int`
+            inside the `test_ds` config.
+
+        Args:
+            outputs: Single or nested list of tensor outputs from one or more data loaders.
+
+        Returns:
+            A dictionary containing the union of all items from individual data_loaders,
+            along with merged logs from all data loaders.
+        """
+        if outputs is not None and len(outputs) == 0:
+            return
+
+        if type(outputs[0]) == dict:
+            return self.multi_validation_epoch_end(outputs, dataloader_idx=0)
+
+        else:
+            output_dict = {'log': {}}
+
+            for dataloader_idx, test_outputs in enumerate(outputs):
+                dataloader_prefix = self.get_test_dataloader_prefix(dataloader_idx)
+                dataloader_logs = self.multi_test_epoch_end(test_outputs, dataloader_idx=dataloader_idx)
+
+                dataloader_logs = dataloader_logs or {}
+
+                for k, v in dataloader_logs.items():
+                    if k == 'log':
+                        log_dict = {}
+                        for k_log, v_log in v.items():
+                            if (
+                                k_log == 'test_loss'
+                                and 'test_loss' not in output_dict['log']
+                                and dataloader_idx == self._test_loss_idx
+                            ):
+                                new_k_log = 'test_loss'
+
+                                # Also insert duplicate key with prefix for ease of comparison
+                                log_dict[dataloader_prefix + k_log] = v_log
+
+                            elif k_log == 'test_loss' and dataloader_idx != self._test_loss_idx:
+                                # replace "test_loss" with <prefix> + loss
+                                # this avoid duplication of the word <prefix> + "test_loss" which causes confusion
+                                new_k_log = dataloader_prefix + 'loss'
+
+                            else:
+                                new_k_log = dataloader_prefix + k_log
+
+                            log_dict[new_k_log] = v_log
+
+                        output_logs = output_dict.get('log', {})
+                        output_logs.update(log_dict)
+
+                        output_dict['log'] = output_logs
+
+                    else:
+                        new_k = dataloader_prefix + k
+                        output_dict[new_k] = v
+
+            return output_dict
+
+    def multi_validation_epoch_end(
+        self, outputs: List[Dict[str, torch.Tensor]], dataloader_idx: int = 0
+    ) -> Optional[Dict[str, Dict[str, torch.Tensor]]]:
+        return
+
+    def multi_test_epoch_end(
+        self, outputs: List[Dict[str, torch.Tensor]], dataloader_idx: int = 0
+    ) -> Optional[Dict[str, Dict[str, torch.Tensor]]]:
+        return
```

Comment:
So if the user overrides validations epoch end, then the entire multi loader support is lost (since they override the ModelPT definition above). In that case this method is never called. But you are right that these should error our in case they only override validation step but don't override multi *. Will raise a NotImplementedError here.